### PR TITLE
Support request compression

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Internal JS-4242

*Description of changes:*
The codegen change to support request compression when trait is available.

Testing done in https://github.com/aws/aws-sdk-js-v3/pull/5625

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (syall) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java:2]: nit: prefer using short-form header with no copyright year

**Review by syall — APPROVED** [review]

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        Map<String, Object> params = new TreeMap<String, Object>();
+
+        // Populate encodings from requestCompression trait
+        RequestCompressionTrait requestCompressionTrait = operation.expectTrait(RequestCompressionTrait.class);
+        params.put("encodings", requestCompressionTrait.getEncodings());
+
+        return params;
+    }
+
+    // return true if operation shape is decorated with `httpChecksum` trait.
```
Comment (kuhe) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java:131]: this should not say httpChecksum, right?

**Review by kuhe — APPROVED** [review]

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java (hunk 1 above)
Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java:2]: Remove Copyright year from header
```suggestion
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Checkstyle comment added in https://github.com/smithy-lang/smithy-typescript/pull/1131

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java (hunk 2 above)
Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java:131]:
```suggestion
    // return true if operation shape is decorated with `requestCompression` trait.
```

**Pull request merged**